packages/hooking/manager.ts

```diff
@@ -180,6 +185,46 @@ export class HookManager {
 				);
 		}
 	}
+
+	registerAfterEachCallback(callback: Thunk) {
+		this.afterEachCallbacks.push(callback);
+	}
+
+	registerBeforeEachCallback(callback: Thunk) {
+		this.beforeEachCallbacks.push(callback);
+	}
+
+	addDictionary(libFuzzerDictionary: string[]) {
+		this.dictionaries.push(this.compileFuzzerDictionary(libFuzzerDictionary));
+	}
+
+	registerInstrumentationPlugin(plugin: () => PluginTarget) {
+		this.instrumentationPlugins.push(plugin);
+	}
+
+	getDictionaries() {
+		return this.dictionaries;
+	}
+
+	getInstrumentationPlugins() {
+		return this.instrumentationPlugins;
+	}
+
+	runAfterEachCallbacks() {
+		for (const afterEachCallback of this.afterEachCallbacks) {
+			afterEachCallback();
+		}
+	}
+
+	runBeforeEachCallbacks() {
+		for (const beforeEachCallback of this.beforeEachCallbacks) {
+			beforeEachCallback();
+		}
+	}
+
+	private compileFuzzerDictionary(lines: string[]): string {
+		return lines.map((line) => `"${line}"`).join("\n");
```

Comment:
`"` and `\` need to be escaped with `\`, see https://llvm.org/docs/LibFuzzer.html#dictionaries